memory/src/main/java/org/opensearch/ml/memory/action/conversation/CreateInteractionTransportAction.java

```diff
@@ -108,4 +110,34 @@ protected void doExecute(Task task, CreateInteractionRequest request, ActionList
         }
     }
 
+    private ActionListener<UpdateResponse> getUpdateResponseListener(
+        String conversationId,
+        String interactionId,
+        ActionListener<CreateInteractionResponse> actionListener
+    ) {
+        return ActionListener.wrap(updateResponse -> {
+            if (updateResponse != null && updateResponse.getResult() == DocWriteResponse.Result.UPDATED) {
+                log
+                    .debug(
+                        "Successfully updated the Conversation with ID: {} after interaction {} is created",
+                        conversationId,
+                        interactionId
+                    );
+                actionListener.onResponse(new CreateInteractionResponse(interactionId));
```

Comment:
[nit] The same line repeated 3 times in the same method. May be we can simplify the code?

Reply:
They are not redundant. They are for different conditions in each branch. This onResponse is used frequently everywhere to send response back to users.

Reply:
I don't understand how `actionListener.onResponse(new CreateInteractionResponse(interactionId));` is different in each branch?

Reply:
They are the same, but each is needed in each branch. The two of them can be moved outside of the if ... else ... and replaced by a single line, but essentially nothing changes. The actionListener.onResponse is required in the error catch anyways.